### PR TITLE
Fixed #33328 -- Transformed formset:added/removed to native JS events.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: django/contrib/admin/static/admin/js/inlines.js [hunk 1]
```diff
@@ -130,7 +135,11 @@
             if (options.removed) {
                 options.removed(row);
             }
-            $(document).trigger('formset:removed', [row, options.prefix]);
+            document.dispatchEvent(new CustomEvent("formset:removed", {
+                detail: {
+                    formsetName: options.prefix
+                }
+            }));
```
Comment (claudep) [inline django/contrib/admin/static/admin/js/inlines.js:142]: I was unusure about triggering the event on the removed row. It looks a bit awkward. Does it make sense to pass to the event a row that is no longer in the DOM?

File: docs/ref/contrib/admin/javascript.txt [hunk 2]
```diff
@@ -72,10 +69,10 @@ listen to the event triggered from there. For example:
 .. code-block:: javascript
     :caption: app/static/app/unregistered_handlers.js
```
Comment (claudep) [inline docs/ref/contrib/admin/javascript.txt:71]: I'm not sure if this entire section about jQuery still makes sense or not.

**Comment by carltongibson** [comment]
buildbot, test on selenium.

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Comment by claudep** [comment]
Thanks Carlton, I'll try to address those until the end of the week.

**Comment by felixxm** [comment]
buildbot, test on selenium.

File: django/contrib/admin/static/admin/js/inlines.js (hunk 1 above)
Comment (matthiask) [reply django/contrib/admin/static/admin/js/inlines.js:142]: I haven't used the row itself in `formset:removed`, ever. Only the prefix. (Both django-admin-ordering and django-content-editor use `formset:added`; I don't have a use for `formset:removed` anymore at all but it was useful in the past.)

**Review by matthiask — APPROVED** [review]
Looks good!

File: docs/ref/contrib/admin/javascript.txt [hunk 3]
```diff
@@ -8,15 +8,15 @@ Inline form events
...
-* ``event`` is the ``jQuery`` event.
-* ``$row`` is the newly added (or removed) row.
-* ``formsetName`` is the formset the row belongs to.
+.. versionchanged:: 4.1
 
-The event is fired using the :ref:`django.jQuery namespace
-<contrib-admin-jquery>`.
+    In older versions, the event was a ``jQuery`` event with ``$row`` and
+    ``formsetName`` parameters. It is now a vanilla JavaScript ``CustomEvent``
+    with parameters set in ``event.detail``.
```
Comment (carltongibson) [inline docs/ref/contrib/admin/javascript.txt]:
```suggestion
    In older versions, the event was a ``jQuery`` event with ``$row`` and
    ``formsetName`` parameters. It is now a JavaScript ``CustomEvent`` with
    parameters set in ``event.detail``.
```

**Comment by carltongibson** [comment]
buildbot, test on selenium.

File: docs/ref/contrib/admin/javascript.txt (hunk 2 above)
Comment (carltongibson) [reply docs/ref/contrib/admin/javascript.txt]: Agreed. It just read at all now. I've removed it.

**Review by carltongibson — APPROVED** [review]
Thanks for this @claudep — I just made small adjustments to the docs, including removing the last paragraphs on jQuery plugins you mentioned. 

👍

File: docs/releases/4.1.txt [hunk 4]
```diff
@@ -411,6 +411,9 @@ Miscellaneous
 
 * The ``type="text/css"`` attribute is no longer included in ``<link>`` tags
   for CSS :doc:`form media </topics/forms/media>`.
+* ``formset:added`` and ``formset:removed`` JavaScript events are now pure
```
Comment (felixxm) [inline docs/releases/4.1.txt:414]: Blank line is missing:
```suggestion

* ``formset:added`` and ``formset:removed`` JavaScript events are now pure
```

File: docs/releases/4.1.txt [hunk 5]
```diff
@@ -411,6 +411,9 @@ Miscellaneous
...
+  JavaScript events and don't depend on jQuery. See
+  :doc:`/ref/contrib/admin/javascript` for more details on the change.
```
Comment (felixxm) [inline docs/releases/4.1.txt:416]: Maybe :thinking: :
```suggestion
  :ref:`admin-javascript-inline-form-events` for more details on the change.
```

**Review by felixxm — APPROVED** [review]
@claudep Thanks :+1:

**Pull request merged**